Tools/autotest/sim_vehicle.py

```diff
@@ -563,14 +563,21 @@ def start_vehicle(binary, autotest, opts, stuff, loc):
         cmd.extend(opts.sitl_instance_args.split(" "))
     if opts.mavlink_gimbal:
         cmd.append("--gimbal")
+    path = None
     if "default_params_filename" in stuff:
         paths = stuff["default_params_filename"]
         if not isinstance(paths, list):
             paths = [paths]
         paths = [os.path.join(autotest, x) for x in paths]
         path = ",".join(paths)
         progress("Using defaults from (%s)" % (path,))
-        cmd.extend(["--defaults", path])
+    if opts.user_param_file:
+        if not os.path.isfile(opts.user_param_file):
+            print("The user param file (%s) does not exist" % (opts.user_param_file,))
+            sys.exit(1)
+        path += "," + str(opts.user_param_file)
+        progress("Adding the user defined params from (%s)" % (str(opts.user_param_file),))
+    cmd.extend(["--defaults", path])
```

Comment:
I think this should check that path is not None, otherwise it's a wrongly formatted input option (SITL might be able to deal with it, but I'm not sure).

Reply:
I could add that, even if I doubt you can do it from sim_vehicle.

Reply:
You certainly can, not all targets have a defaults file.